lib/script/taproot.js

```diff
@@ -0,0 +1,58 @@
+/*!
+* taproot.js - taproot interpreter for bcoin
+* https://github.com/bcoin-org/bcoin
+*/
+
+'use strict';
+
+const bio = require('bufio');
+const common = require('./common');
+const { strcmp } = require('../utils/util');
+
+const {
+    TapLeafHash,
+    TapBranchHash
+} = require('../utils/taggedhash');
+
+const getLeafHash = (script, version) => {
+    const size = script.getVarSize();
+    const bw = bio.write(size + 1);
+    bw.writeU8(version);
+    bw.writeVarBytes(script.raw);
+    const data = bw.render();
+    return TapLeafHash.digest(data);
+};
+
+const taprootTreeHelper = (scripts) => {
```

Comment:
it took me a while to realize `scripts` is an array of `{script, leaf version}` objects. I wonder if we should define a new class `Tapscript` that extends `Script` and adds a leaf version? 

What might be even more prudent is adding a property to `Script`, like `script.tapleafVersion` and default to `null`, but is required for bip342 validation.

Or otherwise at the very least I think we need a js-docs  style comment explaining the input objects.